include/Common/Exceptions.h

```diff
@@ -29,7 +29,10 @@ struct ObjectNotFoundError : virtual ExceptionBase
 {
   const char *what() const noexcept override
   {
-    return "Object not found error";
+    std::string message = "Object not found: ";
+    auto* errinfo = boost::get_error_info<errinfo_object_name>(*this);
```

Comment:
this implies that one can only write `errinfo_object_name` as additional exception info, correct. If this fails, maybe try to extract string?